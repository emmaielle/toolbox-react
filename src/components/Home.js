import React, { Component } from 'react'
import autoBind from 'auto-bind';
import { fetchMirror } from '../actions';
import { withRouter } from 'react-router-dom';
import { Mirror } from './Mirror';
import { Col } from 'react-bootstrap';
import "../styles/containers/Home.css";

export class Home extends Component {
    constructor(props){
        super(props);
        autoBind(this);
    }
    
    handleChangeInput(input) {
        debugger;
        this.props.dispatch(fetchMirror(input));
    }

    render() {
        const { state, props } = this;
        const { mirrorReducer } = props;

        return (
        <Col xs={12} sm={8} md={8} mdOffset={2} smOffset={2} className="home">
            <div className="texts title">Mirror fetcher</div>
            <div className="texts description">
                Enter any text to get the identical text you wrote! 
            </div>
            <Mirror changeInput={this.handleChangeInput} mirrored={mirrorReducer}/>
        </Col>
        )
    }
}

export default withRouter(Home);
