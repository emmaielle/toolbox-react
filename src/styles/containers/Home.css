@keyframes line-scale {
  0% {
    transform: scaley(1); }
  50% {
    transform: scaley(0.4); }
  100% {
    transform: scaley(1); } }

.line-scale > div:nth-child(1) {
  animation: line-scale 1s -0.4s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); }

.line-scale > div:nth-child(2) {
  animation: line-scale 1s -0.3s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); }

.line-scale > div:nth-child(3) {
  animation: line-scale 1s -0.2s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); }

.line-scale > div:nth-child(4) {
  animation: line-scale 1s -0.1s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); }

.line-scale > div:nth-child(5) {
  animation: line-scale 1s 0s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08); }

.line-scale > div {
  background-color: white;
  width: 4px;
  height: 35px;
  border-radius: 2px;
  margin: 2px;
  animation-fill-mode: both;
  display: inline-block; }

.home {
  background-color: #bce9ff;
  min-height: 100vh; }
  .home .texts {
    padding-top: 10px;
    text-align: center; }
    .home .texts.title {
      font-size: 3em; }
  .home .mirror {
    padding-top: 25px;
    text-align: center; }
    .home .mirror .text-area {
      resize: none;
      width: 234px;
      height: 149px;
      border-radius: 8px;
      border: transparent;
      padding: 10px;
      box-shadow: #909090 3px 3px 15px 0px;
      margin: 0px; }
      .home .mirror .text-area:focus {
        outline: none; }
    .home .mirror .result-container {
      font-size: 1.5em; }
      .home .mirror .result-container .success-box {
        padding: 20px; }
        .home .mirror .result-container .success-box .text {
          text-align: justify; }
          .home .mirror .result-container .success-box .text.title {
            font-weight: bold;
            text-decoration: underline; }
  .home .loader-hidden {
    display: none; }
  .home .loader-active {
    display: block; }
