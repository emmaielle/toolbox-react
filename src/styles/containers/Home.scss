$primary-color: white;
@import './node_modules/loaders.css/src/animations/line-scale.scss';

.home {
    background-color: #bce9ff;
    min-height: 100vh;

    .texts {
        padding-top: 10px;
        text-align: center;

        &.title {
            font-size: 3em;
        }
    }

    .mirror {
        padding-top: 25px;
        text-align: center;
        
        .text-area {
            resize: none;
            width: 234px;
            height: 149px;
            border-radius: 8px;
            border: transparent;
            padding: 10px;
            box-shadow: rgb(144, 144, 144) 3px 3px 15px 0px;
            margin: 0px;

            &:focus {
                outline: none;
            }
        }

        .result-container {
            font-size: 1.5em;

            .success-box {
                padding: 20px;

                .text {
                    text-align: justify;
                    
                    &.title {
                        font-weight: bold;
                        text-decoration: underline;
                    }
                }
            }
        }

    }

    .loader-hidden {
        display: none;
    }
    .loader-active {
        display: block;
    }

}
