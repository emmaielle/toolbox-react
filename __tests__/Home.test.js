// MyComponent.test.js
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Home from '../src/components/Home';

describe("Home component", () => {
  it("should render the Home component", () => {
    const wrapper = shallow(<Home />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();

  });
});