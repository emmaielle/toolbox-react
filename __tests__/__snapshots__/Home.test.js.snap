// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render the Home component 1`] = `
<Route>
  <Component />
</Route>
`;
